src/clzones.cpp

```diff
@@ -540,28 +593,41 @@ const zone_manager::zone_data *zone_manager::get_bottom_zone( const tripoint &wh
             return &zone;
         }
     }
+    auto vzones = g->m.get_vehicle_zones( g->u.posz() );
+    for( auto it = vzones.rbegin(); it != vzones.rend(); ++it ) {
+        const auto zone = *it;
 
-    return nullptr;
-}
-
-zone_manager::zone_data &zone_manager::add( const std::string &name, const zone_type_id &type,
-        const bool invert, const bool enabled, const tripoint &start, const tripoint &end,
-        std::shared_ptr<zone_options> options )
-{
-    zones.push_back( zone_data( name, type, invert, enabled, start, end, options ) );
-    cache_data();
+        if( zone->has_inside( where ) ) {
+            return zone;
+        }
+    }
 
-    return zones.back();
+    return nullptr;
 }
 
-bool zone_manager::remove( const size_t index )
-{
-    if( index < zones.size() ) {
-        zones.erase( zones.begin() + index );
-        return true;
+zone_data &zone_manager::add( const std::string &name, const zone_type_id &type,
+                              const bool invert, const bool enabled, const tripoint &start, const tripoint &end,
+                              std::shared_ptr<zone_options> options )
+{
+    zone_data new_zone = zone_data( name, type, invert, enabled, start,
+                                    end, options );
+    //the start is a vehicle tile with cargo space
+    if( const cata::optional<vpart_reference> vp = g->m.veh_at( g->m.getlocal(
+                start ) ).part_with_feature( "CARGO", false ) ) {
+        //TODO:Allow for loot zones on vehicles to be larger than 1x1
+        if( start.x == end.x && start.y == end.y && start.z == end.z ) {
+            //create a vehicle loot zone
+            new_zone.set_is_vehicle( true );
+            vp->vehicle().loot_zones.emplace( vp->mount(), new_zone );
+            g->m.register_vehicle_zone( &vp->vehicle(), g->u.posz() );
```

Comment:
Would be better to use `g->get_levz()` instead of always referring to the player z-level. (Currently this is the same , but it makes no sense why all the zone code needs to use the player position at all.)

Reply:
Thank you! I was actually looking for code that did this but kept looking in "map" instead and never found it >_<

Reply:
The `map` also has this information: `game::get_levz` actually returns `m.get_abs_sub().z`.